src/main/java/org/openpnp/util/Utils2D.java

```diff
@@ -77,43 +77,33 @@ public static Point scalePoint(Point point, double scaleX, double scaleY) {
 		return new Point(point.getX() * scaleX, point.getY() * scaleY);
 	}
 	
-	public static Location calculateBoardPlacementLocation(Location boardLocation, Side boardSide, Location placementLocation) {
+
+	public static Location calculateBoardPlacementLocation(Location boardLocation, double bottom, Location placementLocation) {
         // We will work in the units of the placementLocation, so convert
         // anything that isn't in those units to it.
         boardLocation = boardLocation.convertToUnits(placementLocation.getUnits());
         
         // If we are placing the bottom of the board we need to invert
         // the placement location.
-        if (boardSide == Side.Bottom) {
-            placementLocation = placementLocation.invert(true, false, false, false);
+        if (bottom != 0.0) {
```

Comment:
It's the result of needing board width, and in spirit of the old API i have changed the enum to double.
It have two uses, it hold the board width (X) and at the same time is indicate if side is top or bottom.
If on layout software you place component at 10mm X and 20mm Y, and board width is 100mm,
The true coordinate for placing bottom side is X90 Y20 if the board don't have rotation.
Without having access to board size, it's not possible to place bottom from board-0 origin.
It could be more elegant instead of giving location of Board, Boardlocation is giving.
Having double is maybe more flexible, because the same need to be done on Panels,
and if converting position from camera or teaching, there is no BoardLocation available